js/app/controllers/calcontroller.js

```diff
@@ -140,17 +141,27 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 				$scope.$apply();
 			});
 		} else {
-			$scope.calendarsPromise = CalendarService.getPublicCalendar(constants.publicSharingToken).then(function(calendar) {
-				$scope.calendars = [calendar];
+			$scope.calendars = [];
+			var tokens = new Set(constants.publicSharingToken.split("."));
+			var promises = [];
+			tokens.forEach(function(token){
+				promises.push(CalendarService.getPublicCalendar(token).then(function(calendar) {
+					$scope.calendars.push(calendar);
+				}).catch((reason => {
+					OC.Notification.showTemporary(t('calendar', 'Calendar does not exist'));
+				})));
+			});
+			$scope.calendarsPromise = Promise.all(promises).finally(() => {
```

Comment:
Please use `.then().catch().then()` instead of `finally`.

`finally` has a rather poor browser-support: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#Browser_compatibility